pkg/registries/docker/docker.go

```diff
@@ -182,6 +173,27 @@ func (r *Registry) Match(image *storage.ImageName) bool {
 	return r.repositoryList.Contains(image.GetRemote())
 }
 
+// lazyLoadRepoList will perform the initial repo list load if neccessary.
+// This is safe to call multiple times.
+func (r *Registry) lazyLoadRepoList() {
+	r.repoListOnce.Do(func() {
+		if r.cfg.DisableRepoList {
+			return
+		}
```

Comment:
We should just assume we won't get here until this is false. Based on current usage, that's already the case

Reply:
https://github.com/stackrox/stackrox/pull/11186/commits/8b8ad991e290224d423a05ab23575fa59c6f4df4